### PR TITLE
Bug 1950653: RHDEVDOCS-3033: Add link from build topic to Docker build reference topic

### PR DESCRIPTION
**Pull request by rolfedh**
- Aligned team: Dev Tools
- For branches: 4.5, 4.6, 4.7, 4.8
- https://issues.redhat.com/browse/RHDEVDOCS-3033
- Direct link to doc preview: 
-- https://deploy-preview-32982--osdocs.netlify.app/openshift-enterprise/latest/cicd/builds/understanding-image-builds.html#builds-strategy-docker-build_understanding-image-builds
-- https://deploy-preview-32982--osdocs.netlify.app/openshift-enterprise/latest/cicd/builds/build-strategies.html#builds-strategy-docker-build-arguments_build-strategies
- SME review:  @adambkaplan 
- QE review: @xiuwang 
- Peer review: Oss Tikhomirova
- All reviews completed and approved. Ready to merge.

### CODE REVIEW

File: modules/builds-strategy-docker-build.adoc [hunk 1]
```diff
@@ -6,4 +6,4 @@
 [id="builds-strategy-docker-build_{context}"]
 = Docker build
 
-The docker build strategy invokes the docker build command, and it expects a repository with a Dockerfile and all required artifacts in it to produce a runnable image.
+{product-title} uses buildah to build a container image from a Dockerfile. For more information on building container images with Dockerfiles, see link:https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact[the Dockerfile reference documentation].
```
Comment (adambkaplan) [inline modules/builds-strategy-docker-build.adoc:9]: Link shouldn't jump to the arg/from anchor:

```suggestion
{product-title} uses buildah to build a container image from a Dockerfile. For more information on building container images with Dockerfiles, see link:https://docs.docker.com/engine/reference/builder/[the Dockerfile reference documentation].
```

**Review by adambkaplan — CHANGES_REQUESTED** [review]

**Comment by gabemontero** [comment]
/retitle Bug 1950653: RHDEVDOCS-3033: Add link from build topic to Docker build reference topic

File: modules/builds-strategy-docker-build.adoc [hunk 2]
```diff
@@ -6,4 +6,4 @@
 [id="builds-strategy-docker-build_{context}"]
 = Docker build
 
-The docker build strategy invokes the docker build command, and it expects a repository with a Dockerfile and all required artifacts in it to produce a runnable image.
+{product-title} uses Buildah to build a container image from a Dockerfile. For more information on building container images with Dockerfiles, see link:https://docs.docker.com/engine/reference/builder/[the Dockerfile reference documentation].
```
Comment (gabemontero) [inline modules/builds-strategy-docker-build.adoc:9]: @adambkaplan I'm worried that this is too generic, and we'll miss on the original motivation here, namely the gotcha wrt having to specify the ARG before the FROM.

I get there there are "a bunch of rules" in the referenced docker documentation, and we don't want to repeat what is there, but this specific ARG/FROM thing has come up enough that IMO it needs some highlighting.

Something like:

`For more information on building container images with Dockerfiles, including rules like the ordering of  'ARG'  commands with respect to 'FROM' commands, see.....`

WDYT?

Comment (rolfedh) [reply modules/builds-strategy-docker-build.adoc:9]: @gabemontero @adambkaplan We can do what Gabe proposes, and/or we can add a new subtopic to the [Troubleshooting builds topic](https://deploy-preview-32982--osdocs.netlify.app/openshift-enterprise/latest/cicd/builds/troubleshooting-builds.html).

Comment (rolfedh) [reply modules/builds-strategy-docker-build.adoc:9]: @gabemontero @adambkaplan  I've added the following tip to cover Gabe's suggestion. WDYT?

File: modules/builds-strategy-docker-build.adoc [hunk 3]
```diff
@@ -6,4 +6,9 @@
...
+
+[TIP]
+====
+Also see link:https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact[Understand how ARG and FROM interact] in the Dockerfile reference documentatiosn.
```
Comment (gabemontero) [inline modules/builds-strategy-docker-build.adoc:13]: IMO, the "top level"  link that @adambkaplan had you add make sense there; but this more precise tip needs to context of "I'm manipulating build args".

I would contend this needs to be in the https://docs.openshift.com/container-platform/4.7/cicd/builds/build-strategies.html#builds-strategy-docker-build-arguments_build-strategies section

But maybe @adambkaplan does not agree .... we'll see what he says when he gets a chance to chime in.

Comment (adambkaplan) [reply modules/builds-strategy-docker-build.adoc:13]: I don't think the tip here is necessary. Instead we should link to the Dockerfile `ARG` reference in the build args section of our docs: https://docs.openshift.com/container-platform/4.7/cicd/builds/build-strategies.html#builds-strategy-docker-build-arguments_build-strategies

It looks like that section of the docs has a bad link anyway. The Dockerfile external link goes into the right level of detail wrt to how ARG behaves, and I would rather we don't duplicate content and risk it getting stale over time.

Comment (gabemontero) [reply modules/builds-strategy-docker-build.adoc:13]: OK so @adambkaplan agrees with me @rolfedh about moving this ref/tip to https://docs.openshift.com/container-platform/4.7/cicd/builds/build-strategies.html#builds-strategy-docker-build-arguments_build-strategies

Give that a go please.

**Comment by xiuwang** [comment]
Test the feature and the doc look good to me.

File: modules/builds-strategy-docker-build.adoc (hunk 2 above)
Comment (gabemontero) [reply modules/builds-strategy-docker-build.adoc:9]: works for me

LGTM

@adambkaplan ? ... check out https://github.com/openshift/openshift-docs/pull/32982/files#diff-42a3899e5a04b2d96c9f278774c73d775feffbe8a6c652c72da47e8fb5b9326cR11

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Comment by rolfedh** [comment]
Please note that despite the Chages approved section stating "1 pending reviewer", @xiuwang has already marked this PR as `/label qe-approved`.

File: modules/builds-strategy-docker-build-arguments.adoc [hunk 4]
```diff
@@ -18,3 +23,8 @@ dockerStrategy:
     - name: "foo"
       value: "bar"
 ----
+
+[NOTE]
+====
+Only the `name` and `value` fields are supported. Any settings on the `valueFrom` field are ignoreds.
```
Comment (gabemontero) [inline modules/builds-strategy-docker-build-arguments.adoc:29]: ignoreds -> ignored

Comment (rolfedh) [reply modules/builds-strategy-docker-build-arguments.adoc]: Thanks. Updated.

File: modules/builds-strategy-docker-build-arguments.adoc [hunk 5]
```diff
@@ -6,6 +6,11 @@
 
 You can set link:http://docs.docker.com/v1.7/reference/api/hub_registry_spec/#docker-registry-1-0[docker build arguments] using the `BuildArgs` array. The build arguments are passed to docker when a build is started.
 
+[TIP]
+====
+Consider reviewing link:https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact[Understand how ARG and FROM interact] in the Dockerfile reference documentation.
```
Comment (Preeticp) [inline modules/builds-strategy-docker-build-arguments.adoc:11]: 'Consider reviewing' seems to be similar to using 'please'. How about simply saying saying: Review
Also '...reviewing Understand how ARG and FROM interact...' seems a bit awkward, how about:  'reviewing  the Understand how ARG and FROM interact section ...

File: modules/builds-strategy-docker-build.adoc [hunk 6]
```diff
@@ -6,4 +6,9 @@
 [id="builds-strategy-docker-build_{context}"]
 = Docker build
 
-The docker build strategy invokes the docker build command, and it expects a repository with a Dockerfile and all required artifacts in it to produce a runnable image.
+{product-title} uses Buildah to build a container image from a Dockerfile. For more information on building container images with Dockerfiles, see link:https://docs.docker.com/engine/reference/builder/[the Dockerfile reference documentation].
+
+[TIP]
+====
+If you are working with build arguments, consider reviewing link:https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact[Understand how ARG and FROM interact] in the Dockerfile reference documentation.
```
Comment (Preeticp) [inline modules/builds-strategy-docker-build.adoc:13]: Suggest: If you work... review the link:https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact[Understand how ARG and FROM interact] section...

**Review by Preeticp — COMMENTED** [review]
@rolfedh a couple of minor nits, PTAL and let me know when you are ready to merge.

**Review by ghost — COMMENTED** [review]
LGTM

**Pull request merged**